### PR TITLE
Merge feature/centos into master branch

### PR DESCRIPTION
**Pull request by fyshhh**
This pull request comprises a few commits made to the `feature/centos` branch to support building, launching, checkpointing and restarting MANA on CentOS, as well as some code generalizations. These commits have been tested on Cori and affirmed to work, by comparing the results from `contrib/mpi-proxy-split/test` with and without these commits.

### CODE REVIEW

File: configure-mana [hunk 1]
```diff
@@ -9,18 +9,21 @@
 MPI_ETHERNET_INTERFACE=\
 `ip addr |grep -B1 link/ether | head -1 |sed -e 's%[^ ]*: \([^ ]*\): .*%\1%'`
 
-# This works on our local test computer (Ubuntu-18.04).
+# Note: modify this path according to your environment.
+MPI_INSTALL_DIR=/usr/local
```
Comment (jungan) [inline configure-mana:13]: Suggest check `MPI_INSTALL_DIR` and only assign default value when it was not set.

BTW, do we use `configure-mana` on CORI?

File: contrib/mpi-proxy-split/split_process.cpp [hunk 2]
```diff
@@ -344,21 +344,69 @@ startProxy()
...
+    }
+
+    if (prev_addr_end + 2 * ONEGB <= next_addr_start) {
+      lh_mem_range->start = (VA)prev_addr_end;
+      lh_mem_range->end = (VA)prev_addr_end + 2 * ONEGB;
+      is_set = true;
+      break; 
+    }
+    prev_addr_end = next_addr_end;
+    strncpy(prev_path_name, next_path_name, PATH_MAX - 1);
```
Comment (jungan) [inline contrib/mpi-proxy-split/split_process.cpp:392]: Why do we need to copy `prev_path_name`?

File: include/procmapsarea.h [hunk 3]
```diff
@@ -53,6 +53,11 @@
 #define DELETED_FILE_SUFFIX " (deleted)"
 
 #define FILENAMESIZE        1024
+#define ONEMB (uint64_t)(1024 * 1024)
+#define ONEGB (uint64_t)(1024 * 1024 * 1024)
```
Comment (jungan) [inline include/procmapsarea.h:57]: I don't think we need `uint64_t` here.

File: src/mtcp/mtcp_restart.c [hunk 4]
```diff
@@ -233,8 +234,8 @@ beforeLoadingGniDriverBlockAddressRanges(char *start1, char *end1,
 }
 
 static void
-afterLoadingGniDriverUnblockAddressRanges(char *start1, char *end1,
-	                                  char *start2, char *end2)
+unreserveUpperHalfMemoryRegionsForCkptImgs(char *start1, char *end1,
```
Comment (jungan) [inline src/mtcp/mtcp_restart.c:237]: I think `release` is better than `unreserve`.

File: src/mtcp/mtcp_restart.c [hunk 5]
```diff
@@ -495,13 +496,92 @@ mysetauxval(char **evp, unsigned long int type, unsigned long int val)
...
-#define vdsoStartTmp ((0x40000) - 2*PAGESIZE)
+/* We hardwire these, since we can't make libc calls. */
+#define PAGESIZE 4096
+#define ROUNDADDRUP(addr, size) ((addr + size - 1) & ~(size - 1))
+// We use global variables here, since both remap_vdso_and_vvar_regions and
+// unmap_memory_areas_and_restore_vdso use these variables. They're used to
+// track the temporary regions the vvar and vdso segments are remapped to, so
+// they are not unmapped later.
+uint64_t vvarStartTmp = 0;
+uint64_t vdsoStartTmp = 0;
```
Comment (jungan) [inline src/mtcp/mtcp_restart.c:507]: Add `static` if they will only be used in this file.

File: configure-mana (hunk 1 above)
Comment (fyshhh) [reply configure-mana:13]: yes we do. i've tested this version of `configure-mana` on Cori and it works.

File: contrib/mpi-proxy-split/split_process.cpp (hunk 2 above)
Comment (fyshhh) [reply contrib/mpi-proxy-split/split_process.cpp:392]: this was used in an earlier version of the code, so we don't need to now. i will remove it.

File: include/procmapsarea.h (hunk 3 above)
Comment (fyshhh) [reply include/procmapsarea.h:57]: if this is removed, it causes an assertion error in `mmap.c/getNextAddr`.

File: src/mtcp/mtcp_restart.c (hunk 5 above)
Comment (fyshhh) [reply src/mtcp/mtcp_restart.c:507]: adding `static` strangely causes the `vdso/vvar order was different during ckpt` error.

**Pull request closed without merging**